### ISSUE TITLE
Unresolved dependencies when "exports" field property's value contains directory match before the correct literal match

### ISSUE THREAD

**Issue by chrkulbe** [rollup/plugins#672]
- Rollup Plugin Name: @rollup/plugin-node-resolve
- Rollup Plugin Version: 11
- Rollup Version: 2.34.0
- Operating System (or Browser): Windows 10 x64
- Node Version: v14.15.0
- Link to reproduction _(⚠️ read below)_: https://repl.it/@chrku/rollup-plugin-repro

### Expected Behavior
Successful bundle for packages whose "exports" field in package.json like below:
``` 
"exports": {
    "./helpers/": "./helpers/",
    "./helpers/typeof": "./helpers/typeof.js",
    "./helpers/jsx": "./helpers/jsx.js",
    "./helpers/asyncIterator": "./helpers/asyncIterator.js",
    "./helpers/AwaitValue": "./helpers/AwaitValue.js",
...
    "./helpers/objectWithoutPropertiesLoose": "./helpers/objectWithoutPropertiesLoose.js",
...
    "./helpers/esm/extends": "./helpers/esm/extends.js",
...
}
```

### Actual Behavior
Unable to bundle, output like this:
```
[!] Error: Could not load C:\project\node_modules\react-redux\node_modules\@babel\runtime\helpers\esm\objectWithoutPropertiesLoose (imported by node_modules\react-redux\es\connect\connect.js): ENOENT: no such file or directory, open 'C:\C:\project\node_modules\react-redux\node_modules\@babel\runtime\helpers\esm\objectWithoutPropertiesLoose'
```

### Additional Information
Directory match is preferred over direct file/subpath match, if directory match is configured before subpath match in exports.
This should be a fallback, if no subpatch match exists.

Hint: After removing line 27 - 30 the build was successfully.

https://github.com/rollup/plugins/blob/d35a7eab5fcdab092f13d5d0041363ce12f7611e/packages/node-resolve/src/resolveImportSpecifiers.js#L27-L35

**Comment by LarsDenBakker** [comment]
Looking into this one

**Comment by shellscape** [comment]
Resolved by #675
